src/main/index.js

```diff
@@ -927,6 +927,14 @@ function runApp() {
     return app.getPath('pictures')
   })
 
+  ipcMain.on(IpcChannels.REQUEST_FULLSCREEN, ({ sender }) => {
+    sender.executeJavaScript('document.getElementById("videoContainer").requestFullscreen({navigationUI: "hide"})', true)
+  })
+
+  ipcMain.on(IpcChannels.REQUEST_PIP, ({ sender }) => {
+    sender.executeJavaScript('document.getElementById("video").requestPictureInPicture()', true)
+  })
```

Comment:
Thanks for the suggestion on this - two notes:
- The `requestFullscreen`  callee apparently needs to be the parent of the video element, otherwise it doesn't bind properly with Shaka's fullscreen configurations
- There is [a minute performance benefit to calling `getElementById`](https://stackoverflow.com/a/71878632), albeit very minor

Reply:
> The requestFullscreen callee apparently needs to be the parent of the video element, otherwise it doesn't bind properly with Shaka's fullscreen configurations

We are calling shaka-player's function which does the right thing, it's the same function we call for our `f` hotkey.

Reply:
Sorry to revive an already closed thread. I've changed my mind on this please use the shaka-player methods, that way we know that it will always behave the same. e.g. shaka-player's PiP methods support both normal PiP and document PiP (customisable PiP) but your code only supports the former, so when Electron adds support for document PIP, we'll either have to duplicate all the code for it from shaka-player or call shaka-player's method, which is what I am already asking you to do.